frontend/javascript/basket.js

```diff
@@ -0,0 +1,201 @@
+            .then((data) => {
+                if (getStockAvaiabilityResp(data)) {
+                    increaseQuantity(
+                        decreaseQtyImg,
+                        currentProductId,
+                        currentQuantityElement,
+                        currentUnitCostElement
+                    );
+                } else {
+                    increaseInQuantitybtn.className = 'cart-box-btn__disabled';
+                    increaseInQuantitybtn.setAttribute('title', 'No stock available');
+                }
+            });
+    });
+
+    let decreaseInQuantitybtn = cartBox.querySelector('.cart-box_quantity-decrease');
+    decreaseInQuantitybtn.addEventListener('click', function () {
+        decreaseQuantity(
+            currentQuantityElement,
+            currentProductId,
+            decreaseQtyImg,
+            currentUnitCostElement
+        );
+    });
+}
+
+function getStockAvaiabilityResp(data) {
+    if (data.data.available == true) {
+        return true;
+    }
+}
+
+function isUpdatedQuantityOne(quantity) {
+    return quantity <= 1;
+}
+
+function decrementQuantity(quantity) {
+    return quantity <= 1 ? 1 : quantity - 1;
+}
+
+function increaseQuantity(
+    decreaseQtyImg,
+    currentProductId,
+    currentQuantityElement,
+    currentUnitCostElement
+) {
+    decreaseQtyImg.className = 'cart-box-btn__enabled';
+    let cartItemAsString = JSON.parse(sessionStorage.getItem(currentProductId));
+    let currentQuantity = Number(cartItemAsString.quantity);
+    currentQuantity += 1;
+    let updatedTotalCost = calculateAndSetProductTotalCost(
+        cartItemAsString,
+        currentQuantity,
+        currentProductId
+    );
+    currentUnitCostElement.textContent = '£' + updatedTotalCost;
+    currentQuantityElement.textContent = currentQuantity;
+    updateTotal('increment', currentProductId);
+}
+
+function decreaseQuantity(
+    currentQuantityElement,
+    currentProductId,
+    decreaseQtyImg,
+    currentUnitCostElement
+) {
+    let cartItemAsString = JSON.parse(sessionStorage.getItem(currentProductId));
+    let currentQuantity = Number(cartItemAsString.quantity);
+    const hasMinimumAllowedQuantity = isUpdatedQuantityOne(currentQuantity - 1);
+
+    if (isUpdatedQuantityOne(currentQuantity)) {
+        console.log(currentQuantity);
+        updateTotal('neither', currentProductId);
+    } else {
+        console.log(currentQuantity);
+        updateTotal('decrement', currentProductId);
+    }
+
+    currentQuantity = decrementQuantity(currentQuantity);
+    cartItemAsString.quantity = currentQuantity;
+    let updatedTotalCost = calculateAndSetProductTotalCost(
+        cartItemAsString,
+        currentQuantity,
+        currentProductId
+    );
+    currentUnitCostElement.textContent = '£' + updatedTotalCost;
+    currentQuantityElement.textContent = currentQuantity;
+
+    decreaseQtyImg.className = hasMinimumAllowedQuantity
+        ? 'cart-box-btn__disabled'
```

Comment:
I like the way this has been handled, although I noticed that there's a request being sent when clicking the button even if it has the `cart-box-btn__disabled` class applied. 

I think we talked about the [aria-disabled](https://a11y-101.com/development/aria-disabled) attribute a while ago, one way of handling it could perhaps be to apply that attribute in combination with the class, and either before making the request check whether there's enough quantity, or remove the event listener that triggers the request.